### PR TITLE
Less stringprepping in roster hooks

### PR DESCRIPTION
**Pull request by NelsonVides**
This PR, like several others I've done in the past, aim to reduce the redundancy of stringprepping again and again different blocks of a jid all over the MIM code, and instead taking advantage of the `jid` record. More type-safety, less code, and a tiny slightly bit more performance.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[8980.1](https://travis-ci.org/esl/MongooseIM/jobs/753340789) / Erlang 23.0.3 / small_tests / fe00bff2f784801d91aa20913d5b1e3576626f78
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2997/8980/small_tests.23.0.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2997/8980/small_tests.23.0.3/small/ct_run.mongooseim@localhost.2021-01-07_11.08.37/lib.mongooseim.logs/run.2021-01-07_11.08.38/suite.log.html)

**Comment by mongoose-im** [comment]
[8982.1](https://travis-ci.org/esl/MongooseIM/jobs/753341810) / Erlang 23.0.3 / small_tests / dfab0a31001789bdb0070a59a18df9800de5635a
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2997/8982/small_tests.23.0.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2997/8982/small_tests.23.0.3/small/ct_run.mongooseim@localhost.2021-01-07_11.51.01/lib.mongooseim.logs/run.2021-01-07_11.51.02/suite.log.html)

**Review by janciesla8818 — APPROVED** [review]
Great stuff! Thanks for keeping this coming and improving the performance step by step.

File: src/mod_roster.erl [hunk 1]
```diff
@@ -44,32 +44,45 @@
...
+         get_jid_info/3,
+         remove_user/2, % for tests
+         remove_user/3,
+         get_versioning_feature/2,
+         get_personal_data/2
+        ]).
+
+% Deprecated Hooks
+-export([
+         % get_user_roster/2,
```
[the remark this answers is not in the document]
Comment (NelsonVides) [reply src/mod_roster.erl:76]: Comment extended 😄

**Comment by mongoose-im** [comment]
[9009.1](https://travis-ci.org/esl/MongooseIM/jobs/753976275) / Erlang 23.0.3 / small_tests / ad635686fc22a24e48c0319f61d3d03db8547987
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2997/9009/small_tests.23.0.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2997/9009/small_tests.23.0.3/small/ct_run.mongooseim@localhost.2021-01-11_19.10.53/lib.mongooseim.logs/run.2021-01-11_19.10.53/suite.log.html)

**Pull request merged**